CRM/Contact/Form/Relationship.php

```diff
@@ -590,4 +490,151 @@ public static function getRelationshipTypeMetadata($relationshipList) {
     return $jsData;
   }
 
+  /**
+   * Handling 'delete relationship' action
+   *
+   * @param int $id
+   *   Relationship ID
+   */
+  private function deleteAction($id) {
+    CRM_Contact_BAO_Relationship::del($id);
+
+    // reload all blocks to reflect this change on the user interface.
+    $this->ajaxResponse['reloadBlocks'] = array('#crm-contactinfo-content');
+  }
+
+  /**
+   * Handling updating relationship action
+   *
+   * @param array $params
+   *
+   * @return array
+   */
+  private function updateAction($params) {
+    $params = array_shift($this->preparePostProcessParameters($params));
```

Comment:
This throws a notice error for me -
>Notice: Only variables should be passed by reference in CRM_Contact_Form_Relationship->updateAction() (line 514 of /home/web/civi_master/civicrm/CRM/Contact/Form/Relationship.php).

May be `$this->preparePostProcessParameters($params)` should be assigned to a variable instead of passing directly to `array_shift()` ?